dht_net.go

```diff
@@ -230,31 +247,36 @@ func (ms *messageSender) SendMessage(ctx context.Context, pmes *pb.Message) erro
 			}
 		}
 
-		log.Event(ctx, "dhtSentMessage", ms.dht.self, ms.p, pmes)
-
-		if ms.singleMes > streamReuseTries {
-			ms.s.Close()
-			ms.s = nil
-		} else if retry {
+		if retry {
 			ms.singleMes++
+			if ms.singleMes > streamReuseTries {
+				ms.resetHard()
+			}
 		}
 
 		return nil
 	}
 }
 
 func (ms *messageSender) SendRequest(ctx context.Context, pmes *pb.Message) (*pb.Message, error) {
-	ms.lk.Lock()
-	defer ms.lk.Unlock()
+	defer log.EventBegin(ctx, "dhtSendRequest", ms.dht.self, ms.p, pmes).Done()
```

Comment:
dont technically need to prefix the event type with "dht", the `log` object is created as a logger for the dht subsystem

Reply:
ipfs log only gives you the event name in the `event` field, so that's kind of necessary to disambiguate events.

Reply:
i mean for practical purposes with grep, otherwise the system field does it just fine.

Reply:
I will remove the prefix, the canonical way to process the event log is jq; and grep becomes only slightly more complicated, it can be a double grep for dht and then SendRequest